indexing-service/src/main/java/io/druid/indexing/overlord/supervisor/SupervisorManager.java

```diff
@@ -181,7 +182,13 @@ private boolean createAndStartSupervisorInternal(SupervisorSpec spec, boolean pe
     supervisor.start(); // try starting the supervisor first so we don't persist a bad spec
 
     if (persistSpec) {
-      metadataSupervisorManager.insert(id, spec);
+      try {
+        metadataSupervisorManager.insert(id, spec);
+      } catch (Exception e) {
+        // Got exception while persisting the spec, stop the supervisor and propagate the exception
+        supervisor.stop(false);
```

Comment:
If we stop it gracefully that means tasks that many have been started by this supervisor are allowed to publish what they have read which might be really tiny data as the metadata insert happens just after the supervisor starts. 

Ideally `supervisor.start()` should be called only if metadata insert succeeds but it is not possible here.